applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py

```diff
@@ -134,55 +155,166 @@ def Initialize(self):
 
         print ("Mesh stage solver initialization finished")
 
-    def Solve(self):
-        # Solve equations on mesh
-        (self.solver).Solve()
+    def AdvanceInTime(self, current_time):
+        dt = self._ComputeDeltaTime()
+        new_time = current_time + dt
+
+        self.main_model_part.CloneTimeStep(new_time)
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+
+        return new_time
+
+    def InitializeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.InitializeSolutionStep()
+
+    def Predict(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.Predict()
+
+    def SolveSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            is_converged = self.solver.SolveSolutionStep()
+            return is_converged
+
+    def FinalizeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.FinalizeSolutionStep()
+
+    def Check(self):
+        self.solver.Check()
 
     def Clear(self):
-        (self.solver).Clear()
+        self.solver.Clear()
+
+    def Solve(self):
```

Comment:
I would remove this method completely, it is already in the baseclass including the warning